ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java

```diff
@@ -0,0 +1,291 @@
+     */
+    public void getTraces(String parentInteractionId, ActionListener<List<Interaction>> actionListener) {
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().newStoredContext(true)) {
+            if (!clusterService.state().metadata().hasIndex(indexName)) {
+                actionListener.onResponse(List.of());
+                return;
+            }
+            innerGetTraces(parentInteractionId, actionListener);
+        } catch (Exception e) {
+            actionListener.onFailure(e);
+        }
+    }
+
+    @VisibleForTesting
+    void innerGetTraces(String parentInteractionId, ActionListener<List<Interaction>> listener) {
+        SearchRequest searchRequest = Requests.searchRequest(indexName);
+
+        // Build the query
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery();
+
+        // Add the ExistsQueryBuilder for checking null values
+        ExistsQueryBuilder existsQueryBuilder = QueryBuilders.existsQuery(ConversationalIndexConstants.INTERACTIONS_TRACE_NUMBER_FIELD);
+        boolQueryBuilder.must(existsQueryBuilder);
+
+        // Add the TermQueryBuilder for another field
+        TermQueryBuilder termQueryBuilder = QueryBuilders
+            .termQuery(ConversationalIndexConstants.PARENT_INTERACTIONS_ID_FIELD, parentInteractionId);
+        boolQueryBuilder.must(termQueryBuilder);
+
+        // Set the query to the search source
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(boolQueryBuilder);
+        searchRequest.source(searchSourceBuilder);
+
+        searchRequest.source().sort(ConversationalIndexConstants.INTERACTIONS_TRACE_NUMBER_FIELD, SortOrder.ASC);
+        try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+            ActionListener<List<Interaction>> internalListener = ActionListener.runBefore(listener, () -> threadContext.restore());
+            ActionListener<SearchResponse> al = ActionListener.wrap(response -> {
+                List<Interaction> result = new LinkedList<Interaction>();
+                for (SearchHit hit : response.getHits()) {
+                    result.add(Interaction.fromSearchHit(hit));
+                }
+                internalListener.onResponse(result);
+            }, e -> { internalListener.onFailure(e); });
+            client
+                .admin()
+                .indices()
+                .refresh(Requests.refreshRequest(indexName), ActionListener.wrap(r -> { client.search(searchRequest, al); }, e -> {
+                    internalListener.onFailure(e);
+                }));
+        } catch (Exception e) {
+            listener.onFailure(e);
+        }
+    }
+
+    /**
+     * Get the interactions associate with this conversation, sorted by recency
+     * @param interactionId the parent interaction id whose traces to get
+     * @param actionListener listener for the update response
```

Comment:
add `updateContent`?

Reply:
I'll remove this class due to a new one in Xun's pr.